platform/mellanox/mlnx-fw-upgrade.j2

```diff
@@ -250,22 +266,46 @@ function UpgradeFWFromImage() {
     fi
 }
 
+function UpgradeFWFromFile() {
+    local -r _FW_USER_FILE="/host/mlnx/asic/fw.mfa"
+
+    if [ ! -f "${_FW_USER_FILE}" ]; then
+        LogWarning "User defined ASIC firmware file not installed"
+        ExitFailure "Failed to upgrade ASIC firmware"
+    fi
+
+    UpgradeFW --from-file "${_FW_USER_FILE}"
+}
+
 function ExitIfQEMU() {
     if [ -n "$(lspci -vvv | grep SimX)" ]; then
         ExitSuccess "No FW upgrade for SimX platform"
     fi
 }
 
+function ExitIfAutoUpdateDisabled() {
+    local -r AUSTATE="$(sonic-cfggen -d -v FIRMWARE[\'asic\'][\'auto_update\'])"
+    if [[ "${AUSTATE}" == "disable" ]]; then
+        ExitSuccess "FW auto update disabled. Firmware won't be upgrade"
+    fi
+}
+
 ParseArguments "$@"
 
 ExitIfQEMU
+ExitIfAutoUpdateDisabled
 
 WaitForDevice
 
-if [ "${IMAGE_UPGRADE}" != "${YES_PARAM}" ]; then
-    UpgradeFW
+DEFAULT_SOURCE="$(sonic-cfggen -d -v FIRMWARE[\'asic\'][\'default\'])"
```

Comment:
Also please make sure that we are able to perform reboot when database service is down or config db is not initialized